### PR TITLE
Add two-phase material parameter input for heat transfer

### PR DESCRIPTION
**Pull request by nmuch**
This PR introduces a two-phase material parameter input section for heat transfer. This material parameter section can be expanded for more material parameters in other physical contexts.

@mschreter what do you think of this design? If we always use the material section to define materials we can remove the material_data pointer and pass this struct by reference instead.

This might create a merge conflict with #106. I'll rebase whatever is merged last.

### CODE REVIEW

File: include/meltpooldg/simulations/unidirectional_heat_transfer/unidirectional_heat_transfer_with_two_phase.json [hunk 1]
```diff
@@ -25,9 +22,16 @@
...
-    "paraview print boundary id": "true",
-    "paraview filename": "solution_heat_transfer_two_phase"
+  "material": {
+    "material liquid capacity": "460",
+    "material liquid conductivity": "55.563",
+    "material liquid density": "7850",
+    "material gas capacity": "230",
+    "material gas conductivity": "27.7815",
+    "material gas density": "3925",
+    "paraview": {
```
Comment (mschreter) [inline include/meltpooldg/simulations/unidirectional_heat_transfer/unidirectional_heat_transfer_with_two_phase.json:32]: I think there is a bracelet missing?

File: include/meltpooldg/interface/parameters.hpp [hunk 2]
```diff
@@ -237,6 +237,24 @@ namespace MeltPoolDG
     number ls_value_gas                       = -1.0;
   };
 
+  template <typename number = double>
+  struct MaterialData
+  {
+    struct Liquid
+    {
+      number capacity;
```
Comment (mschreter) [inline include/meltpooldg/interface/parameters.hpp:245]: Would you mind to provide default values to the parameters?

**Comment by mschreter** [comment]
Thank you Nils, its a good start for the moment. Just to briefly rephrase what we've discussed, in the future we might think about having a `material` object, holding parameters of only one material or also many more phases/materials, which can be accessed by `get_phase_properties("phase_1")` or similar. I will have a look at the tests also from the other PR and will then merge one PR by another.

**Comment by mschreter** [comment]
@nmuch: I simply used the online tool for fixing the merge conflicts. Maybe you can address the minor comments I had and in the meantime I will again have a look at the tests. Thanks a lot, I think we made nice progress this week :+1:.

File: include/meltpooldg/interface/parameters.hpp (hunk 2 above)
Comment (nmuch) [reply include/meltpooldg/interface/parameters.hpp:246]: Sure

**Pull request merged**